src/Client.php

```diff
@@ -337,6 +337,26 @@ public function waitFor(string $locator, int $timeoutInSecond = 30, int $interva
         return $this->crawler = $this->createCrawler();
     }
 
+    /**
+     * @param string $locator
+     * @param int $timeoutInSecond
+     * @param int $intervalInMillisecond
+     * @throws NoSuchElementException
+     * @throws TimeoutException
+     */
+    public function waitForVisibility(string $locator, int $timeoutInSecond = 30, int $intervalInMillisecond = 250)
+    {
+        $locator = trim($locator);
+
+        $by = '' === $locator || '/' !== $locator[0]
+            ? WebDriverBy::cssSelector($locator)
+            : WebDriverBy::xpath($locator);
+
+        $this->wait($timeoutInSecond, $intervalInMillisecond)->until(
+            WebDriverExpectedCondition::visibilityOfElementLocated($by)
+        );
+    }
```

Comment:
Does this method need to return an updated `Crawler` instance as is the case with `waitFor()`?

Reply:
I think is not necessary but i updated the function to return the updated `Crawler` .

Reply:
Just to be clear, I wasn't suggesting that was necessary.

 I noticed there was a discrepancy between two methods that were almost identical.

Reply:
But if it is acceptable for this method to return the updated crawler, there is now just a single line difference between `waitForVisibility()` and `waitFor()` which could probably be further factored out.